databroker/eventsource/archiver.py

```diff
@@ -10,45 +10,113 @@
 import pytz
 import uuid
 
+import numpy as np
+
 import requests
 import pandas as pd
 
+from ..utils import ALL
 
 class ArchiverEventSource(object):
-    def __init__(self, url, timezone, pvs):
+    
+    def __init__(self, config):
         """
-        DataBroker plugin
-
-        Parameters
-        ----------
-        url : string
-            e.g., 'http://host:port/'
-        timezone : string
-            e.g., 'US/Eastern'
-        pvs : dict
-            a dict mapping human-friendly names to PVs
+        Shim class to turn the EPICS Archiver Appliance into EventSource
         """
+
+        if all (k in config for k in ['name', 'url']):
+            pass
+        else:
+            raise TypeError("config {} does not include one of required"
+                            " keys (name, url).".format(config))
+        
+        self._name = config['name']
+        
+        url =  config['url']
         if not url.endswith('/'):
             url += '/'
         self.url = url
         self.archiver_addr = self.url + "retrieval/data/getData.json"
+
+        if 'timezone' in config:
+            timezone = config['timezone']
+        else:
+            timezone = 'US/Eastern'
         self.tz = pytz.timezone(timezone)
+
+        if 'pvs' in config:
+            pvs = config['pvs']
+        else :
+            pvs = {}
         self.pvs = pvs
+        
         self._descriptors = {}
 
+
+    @property
+    def name(self):
+        return self._name
+
+    def add_pvs(self, new_pvs):
```

Comment:
There is something to be said for returning a copy so that the user can not accidentally mutate it someplace else in the code, but `es.pvs.update({...})` is _also_ very nice.  I am not sure of anywhere else we go to such lengths to protect the user from them selves (and in fact in the case of `read_attrs` we go to great lengths to fake returning a mutable thing!).

Reply:
We also had a very nasty bug with the hkl library where we were doing

```python
config = HKLEngine.get_config()
config.set_stuff()
```

but the object returned was a copy (not a proxy) for the underlying c++ object so various constraints were not actually being set so the computations were wrong.  This API has echos of that.

Reply:
@malitsky Do you agree then that `add_pvs` and `get_pvs` should be removed?

Reply:
As already explained in the documentation’s comments, add_pvs and get_pvs provide a user with the optimal mechanism for extending the Databroker configuration file with application-specific pvs (note: beamline archivers maintain tens of thousands pvs):

https://github.com/malitsky/databroker/blob/master/doc/source/archiver.rst

Let me know when you plan to merge this PR.

Reply:
These methods follow a conventional approach for separating the class interface from its members. The optimal interface however is a trade-off among objective and subjective factors, so these methods per se can be removed.